### PR TITLE
Table component branch bug

### PR DESCRIPTION
**Pull request by kiraarghy**
Markdown example to demonstratte indentation problem with prettier and marksy.

In this [case](https://github.com/FormidableLabs/spectacle/pull/746/files#diff-27d1398137d07ae13f238fccada6a73fR90-R101) marksy interprets the code as a codepane, not a table. After having it explained by Carlos, this makes sense but this is probably a sub-optimal user experience and we should think about how to solve this.

### CODE REVIEW

**Comment by kitten** [comment]
It’s be great to just apply a simple strip-indent to the content of the markdown code, similar to: https://github.com/sindresorhus/strip-indent

File: examples/js/test-js.js [hunk 1]
```diff
@@ -87,6 +87,18 @@ const TestJs = () => (
     )}
   >
     <Slide>
+      <Slide>
+        <Markdown>{`
+        # hello there
+        
+        | hi   | Supported | Versions |
+|-----------|-----------|----------|
```
Comment (carlos-kelly) [inline examples/js/test-js.js]: This indentation looks strange.

**Comment by kiraarghy** [comment]
Oh I have a separate branch that I was going to have a look at the styling in, will get that up soon!

**Comment by kiraarghy** [comment]
> It’s be great to just apply a simple strip-indent to the content of the markdown code, similar to: https://github.com/sindresorhus/strip-indent

I'm not sure if there is a simple solution to this as @carlos-kelly and @ryan-roemer have said if I want to put spacing in my markdown if we strip out all indent it would screw up that spacing? 

I'm not sure if this would solve the issue highlighted [here](https://github.com/FormidableLabs/spectacle/pull/746/files#diff-27d1398137d07ae13f238fccada6a73fR90-R101) as the least number of leading whitespace is line 95?

File: examples/js/test-js.js (hunk 1 above)
Comment (kiraarghy) [reply examples/js/test-js.js]: Yeah it's wrong, it's just not immediately noticeable that it's wrong 😭

**Comment by ryan-roemer** [comment]
As an FYI `strip-indent` is exactly the same in function as our indent normalization function, it just uses two packages (`min-indent` and `strip-indent`). We're likely going to need something different.

File: examples/js/test-js.js (hunk 1 above)
Comment (ryan-roemer) [reply examples/js/test-js.js]: For this specific case, I don't see how we're going to handle it outside of something super complicated like probablistically calculating indent based on markdown parsing.

The only idea here I can come up with is "indent cannot be `0`" and ignore a detected min indent that is `0` relying on everything else....

**Comment by ryan-roemer** [comment]
The original example had incorrect `<Slide>` nesting that threw everything off. I've:

- Fixed the nesting
- Moved the slide to the markdown examples section with others
- Updated `.prettierignore` with our current outputs
- Added a comparable `one-page.html` table slide

With this commit, it appears that just doing something like:

```js
    <Slide>
      <Markdown>
        {`
          # hello there

          | hi              | Supported | Versions |
          |-----------------|-----------|----------|
          | Chrome          | Yes       | Last 2   |
          | Firefox         | Yes       | Last 2   |
          | Opera           | Yes       | Last 2   |
          | Edge (EdgeHTML) | No        |          |
          | IE 11           | No        |          |
        `}
      </Markdown>
    </Slide>
```

Works fine. We still need to do the nested `{\`STRING\`}` thing, but maybe that's not too bad?

And, I've run everything through `yarn prettier-fix` and I don't see prettier doing any further mutations or changes to the table markdown string.

**Comment by kiraarghy** [comment]
@ryan-roemer that's amazing, thanks!

**Pull request merged**